pwiz_tools/Skyline/TestData/CommandLineTest.cs

```diff
@@ -289,6 +289,40 @@ public void ConsoleAddFastaTest()
             AssertEx.IsDocumentState(doc, 0, 2, 7, 7, 49);
         }
 
+        [TestMethod]
+        public void ConsoleNewDocumentTest()
+        {
+            TestFilesDir = new TestFilesDir(TestContext, ZIP_FILE);
+            string docPath = TestFilesDir.GetTestPath("BSA_Protea_label_free_20100323_meth3_multi.sky");
+            string outPath = TestFilesDir.GetTestPath("AddFasta_Out.sky");
+            string fastaPath = TestFilesDir.GetTestPath("sample.fasta");
+
+            // arguments that would normally be quoted on the command-line shouldn't be quoted here
+            var settings = new[]
+            {
+                "--new=" + docPath,
+                "--full-scan-precursor-isotopes=Count",
+                "--full-scan-acquisition-method=DDA",
+                "--tran-precursor-ion-charges=2,3,4",
+                "--tran-product-ion-charges=1,2",
+                "--tran-product-start-ion=ion 1",
+                "--tran-product-end-ion=last ion - 1"
+            };
+
+            string output = RunCommand(settings);
+            StringAssert.Contains(output, string.Format(Resources.CommandLine_NewSkyFile_FileAlreadyExists, docPath));
+
+            output = RunCommand(settings.Append("--overwrite").ToArray());
+            StringAssert.Contains(output, string.Format("Deleting existing file '{0}'", docPath));
+            AssertEx.DoesNotContain(output, Resources.CommandLineTest_ConsoleAddFastaTest_Error);
+            AssertEx.DoesNotContain(output, Resources.CommandLineTest_ConsoleAddFastaTest_Warning);
+
+            SrmDocument doc = ResultsUtil.DeserializeDocument(docPath);
+            Assert.AreEqual(FullScanPrecursorIsotopes.Count, doc.Settings.TransitionSettings.FullScan.PrecursorIsotopes);
+            Assert.AreEqual(FullScanAcquisitionMethod.DDA, doc.Settings.TransitionSettings.FullScan.AcquisitionMethod);
+            Assert.AreEqual("2, 3, 4", doc.Settings.TransitionSettings.Filter.PeptidePrecursorChargesString);
+        }

```

Comment:
Yeah I made that test before I started working on really making the CLI outcome for the DiaSwath tests the same as the GUI. That required adding a lot more parameters which I didn't get around to adding to the functional test yet. Are you suggesting a functional test that is more comprehensive than just being sure that all the parameters work and change the settings they're supposed to?

Reply:
I'm afraid I don't understand most of that comment. :( What doesn't work that way with SkylineCmd? What ways of running the command-line are we talking about? SkylineCmd vs. SkylineRunner?

Reply:
The file that "pwiz.Skyline.Properties.Settings" gets read from depends on the executing assembly.
Skyline SkylineCmd.exe runs Skyline, the executing assembly is SkylineCmd.exe.

When SkylineRunner.exe runs Skyline, SkylineRunner finds the Skyline executable on the Start menu, and runs Skyline.exe in a separate process, which means that it's using the exact same settings as when you run the ordinary user interface.

(I am not sure exactly how .Net decides where to load settings from, but it definitely depends on the executing assembly, which is why when you run TestRunner.exe, Skyline ends up loading its settings from a completely different place).

Reply:
OK, now I get it. Yeah, sharing settings when the exe name changes would be tricky. Let me see if that can be overridden.

Reply:
I/we didn't know how to short circuit the GUI and attach to the console that Skyline was launched from.

The only thing we knew how to do was to tell Visual Studio what type of program we had by choosing either "Windows Application" or "Console Application" in the "Output type" dropdown in Visual Studio Application Properties.

We had been figuring that there was no other way to accomplish this based on the fact that Java has always required separate executables "java.exe" and "javaw.exe" but yes it would make some things a lot easier if we could find out that we were wrong about this.

Reply:
It is fairly trivial to attach a WinForms app to the [parent's console ](https://learn.microsoft.com/en-us/windows/console/attachconsole) so that Console.WriteLine will show up in the console when it is launched from a console. I'm not sure if it's exactly the same as making it a "Console Application", but it's pretty close:

> This function is primarily useful to applications that were linked with [/SUBSYSTEM:WINDOWS](https://learn.microsoft.com/en-us/cpp/build/reference/subsystem-specify-subsystem), which implies to the operating system that a console is not needed before entering the program's main method. In that instance, the standard handles retrieved with [GetStdHandle](https://learn.microsoft.com/en-us/windows/console/getstdhandle) will likely be invalid on startup until AttachConsole is called. The exception to this is if the application is launched with handle inheritance by its parent process.

IIRC I added AttachConsole for SeeMS when I wanted to see errors in the console when launching it from command-line.

Reply:
Oh, if we did merge SkylineCmd with Skyline it would also avoid the circular dependency issue with the installer.